packages/dapp/src/components/DualWidget/WidgetEvents.tsx

```diff
@@ -112,9 +129,32 @@ export function WidgetEvents() {
     );
     widgetEvents.on(WidgetEvent.RouteExecutionFailed, onRouteExecutionFailed);
     widgetEvents.on(WidgetEvent.RouteHighValueLoss, onRouteHighValueLoss);
+    widgetEvents.on(WidgetEvent.SafeRouteInitiated, onSafeRouteInitiation);
 
     return () => widgetEvents.all.clear();
   }, [trackAttribute, trackEvent, trackTransaction, widgetEvents]);
 
-  return null;
+  const handleMultiSigConfirmationModalClose = () => {
+    setIsMultiSigConfirmationModalOpen(false);
+  };
+
+  useEffect(() => {
+    const isSafeSigner = !!(account?.signer?.provider as any)?.provider?.safe
+      ?.safeAddress;
+
+    setIsMultisigConnectedAlertOpen(isSafeSigner);
+  }, [account.address]);
+
+  return (
+    <>
+      <MultisigConnectedAlert
+        open={isMultisigConnectedAlertOpen}
+        onClose={() => setIsMultisigConnectedAlertOpen(false)}
+      />
+      <MultisigConfirmationModal
+        open={isMultiSigConfirmationModalOpen}
+        onClose={handleMultiSigConfirmationModalClose}
```

Comment:
We should stick to one way of `onClose` handling, preferably without inline functions.

Reply:
Got it. Sure, I'll fix it